torchgeo/datasets/sentinel.py

```diff
@@ -95,11 +95,11 @@ def __init__(
         Raises:
             FileNotFoundError: if no files are found in ``root``
         """
-        self.bands = bands if bands else self.all_bands
-        self.filename_glob = self.filename_glob.format(self.bands[0])
+        bands = bands or self.all_bands
```

Comment:
I like what you did here, can we do the same thing for landsat too? I think the situation is the same, but this logic is much more clear than the if-statement.